Add mcxa166 mcxa276 lptmr ctimer

peterwangsz opened the pull request:
enable the lptmr and ctimer with zephyr counter driver
verified tests/drivers/count/counter_basic_api

peterwangsz committed:
boards: frdm_mcxa166, frdm_mcxa276: add lptmr and ctimer support

1. enable lptmr and ctimer support
2. verified tests/drivers/count/counter_basic_api

Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

hakehuang commented:
board testing pass
```
*** Booting Zephyr OS build v4.1.0-2646-ge41909a32ce9 ***

Printing everything in utsname...
sysname[7]: Zephyr
n*** Booting Zephyr OS build v4.1.0-2693-g3f79675eb0eb ***
Running TESTSUITE counter_basic
===================================================================
START - test_all_channels
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_all_channels in 0.235 seconds
===================================================================
START - test_cancelled_alarm_does_not_expire
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_cancelled_alarm_does_not_expire in 0.205 seconds
===================================================================
START - test_late_alarm
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_late_alarm in 0.205 seconds
===================================================================
START - test_late_alarm_error
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_late_alarm_error in 0.205 seconds
===================================================================
START - test_multiple_alarms
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_multiple_alarms in 0.259 seconds
===================================================================
START - test_set_top_value_with_alarm
Testing ctimer@40004000
Testing lptmr@400ab000
 PASS - test_set_top_value_with_alarm in 0.423 seconds
===================================================================
START - test_short_relative_alarm
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_short_relative_alarm in 0.206 seconds
===================================================================
START - test_single_shot_alarm_notop
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_single_shot_alarm_notop in 0.275 seconds
===================================================================
START - test_single_shot_alarm_top
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_single_shot_alarm_top in 0.275 seconds
===================================================================
START - test_valid_function_without_alarm
Testing ctimer@40004000
Testing lptmr@400ab000
 PASS - test_valid_function_without_alarm in 0.268 seconds
===================================================================
TESTSUITE counter_basic succeeded
Running TESTSUITE counter_no_callback
===================================================================
START - test_set_top_value_without_alarm
Testing ctimer@40004000
Testing lptmr@400ab000
 PASS - test_set_top_value_without_alarm in 0.215 seconds
===================================================================
TESTSUITE counter_no_callback succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [counter_basic]: pass = 6, fail = 0, skip = 4, total = 10 duration = 2.556 seconds
 - PASS - [counter_basic.test_all_channels] duration = 0.235 seconds
 - SKIP - [counter_basic.test_cancelled_alarm_does_not_expire] duration = 0.205 seconds
 - SKIP - [counter_basic.test_late_alarm] duration = 0.205 seconds
 - SKIP - [counter_basic.test_late_alarm_error] duration = 0.205 seconds
 - PASS - [counter_basic.test_multiple_alarms] duration = 0.259 seconds
 - PASS - [counter_basic.test_set_top_value_with_alarm] duration = 0.423 seconds
 - SKIP - [counter_basic.test_short_relative_alarm] duration = 0.206 seconds
 - PASS - [counter_basic.test_single_shot_alarm_notop] duration = 0.275 seconds
 - PASS - [counter_basic.test_single_shot_alarm_top] duration = 0.275 seconds
 - PASS - [counter_basic.test_valid_function_without_alarm] duration = 0.268 seconds

SUITE PASS - 100.00% [counter_no_callback]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.215 seconds
 - PASS - [counter_no_callback.test_set_top_value_without_alarm] duration = 0.215 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
RunID: 310619b7aa1a8dae1e785c57f0f31ebe
PROJECT EXECUTION SUCCESSFUL

```

hakehuang commented:
and frdm_mcxa276

```
*** Booting Zephyr OS build v4.1.0-2693-g3f79675eb0eb ***
Running TESTSUITE counter_basic
===================================================================
START - test_all_channels
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_all_channels in 0.235 seconds
===================================================================
START - test_cancelled_alarm_does_not_expire
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_cancelled_alarm_does_not_expire in 0.205 seconds
===================================================================
START - test_late_alarm
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_late_alarm in 0.205 seconds
===================================================================
START - test_late_alarm_error
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_late_alarm_error in 0.205 seconds
===================================================================
START - test_multiple_alarms
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_multiple_alarms in 0.259 seconds
===================================================================
START - test_set_top_value_with_alarm
Testing ctimer@40004000
Testing lptmr@400ab000
 PASS - test_set_top_value_with_alarm in 0.423 seconds
===================================================================
START - test_short_relative_alarm
Skipped for ctimer@40004000
Skipped for lptmr@400ab000
 SKIP - test_short_relative_alarm in 0.206 seconds
===================================================================
START - test_single_shot_alarm_notop
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_single_shot_alarm_notop in 0.275 seconds
===================================================================
START - test_single_shot_alarm_top
Testing ctimer@40004000
Skipped for lptmr@400ab000
 PASS - test_single_shot_alarm_top in 0.275 seconds
===================================================================
START - test_valid_function_without_alarm
Testing ctimer@40004000
Testing lptmr@400ab000
 PASS - test_valid_function_without_alarm in 0.268 seconds
===================================================================
TESTSUITE counter_basic succeeded
Running TESTSUITE counter_no_callback
===================================================================
START - test_set_top_value_without_alarm
Testing ctimer@40004000
Testing lptmr@400ab000
 PASS - test_set_top_value_without_alarm in 0.215 seconds
===================================================================
TESTSUITE counter_no_callback succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [counter_basic]: pass = 6, fail = 0, skip = 4, total = 10 duration = 2.556 seconds
 - PASS - [counter_basic.test_all_channels] duration = 0.235 seconds
 - SKIP - [counter_basic.test_cancelled_alarm_does_not_expire] duration = 0.205 seconds
 - SKIP - [counter_basic.test_late_alarm] duration = 0.205 seconds
 - SKIP - [counter_basic.test_late_alarm_error] duration = 0.205 seconds
 - PASS - [counter_basic.test_multiple_alarms] duration = 0.259 seconds
 - PASS - [counter_basic.test_set_top_value_with_alarm] duration = 0.423 seconds
 - SKIP - [counter_basic.test_short_relative_alarm] duration = 0.206 seconds
 - PASS - [counter_basic.test_single_shot_alarm_notop] duration = 0.275 seconds
 - PASS - [counter_basic.test_single_shot_alarm_top] duration = 0.275 seconds
 - PASS - [counter_basic.test_valid_function_without_alarm] duration = 0.268 seconds

SUITE PASS - 100.00% [counter_no_callback]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.215 seconds
 - PASS - [counter_no_callback.test_set_top_value_without_alarm] duration = 0.215 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
RunID: 3b45dad92bb562b677a31332e01b104d
PROJECT EXECUTION SUCCESSFUL

```

mmahadevan108 commented on boards/nxp/frdm_mcxa276/board.c:
```diff
@@ -15,12 +15,23 @@
 	spc_sram_voltage_config_t sramOption;
 
+	/* Enable LPCAC */
+	enable_lpcac();
```
This should be in a separate commit with some details about enabling the cache

peterwangsz replied:
accept and updated to new commit, please help to review again.
thanks.

mmahadevan108 commented on boards/nxp/frdm_mcxa166/board.c:
```diff
@@ -15,12 +15,23 @@
 	spc_sram_voltage_config_t sramOption;
 
+	/* Enable LPCAC */
+	enable_lpcac();
```
This should be in a separate commit with some details about enabling the cache

mmahadevan108 commented on boards/nxp/frdm_mcxa166/board.c:
```diff
@@ -15,12 +15,23 @@
+	SYSCON->LPCAC_CTRL &=
+		~(SYSCON_LPCAC_CTRL_CLR_LPCAC_MASK | SYSCON_LPCAC_CTRL_DIS_LPCAC_MASK);
+}
+
```
This is already done in the HAL as part of `SystemInit`. I don't think this is needed.

peterwangsz commented:
Please help to review. thanks.

peterwangsz committed:
boards: frdm_mcxa166, frdm_mcxa276: fix SYS_CLOCK_HW_CYCLES_PER_SEC

1. mcxa serial SYS_CLOCK_HW_CYCLES_PER_SEC default value is 96000000
2. mcxa166,mcxa276 works at 180000000

Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

peterwangsz commented on boards/nxp/frdm_mcxa166/frdm_mcxa166_defconfig:
```diff
@@ -9,3 +9,4 @@ CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_GPIO=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=180000000
```
updated as suggestion, please help to review. thanks.

dleach02 reviewed (dismissed):
lgtm... need to address merge conflicts

hakehuang reviewed (dismissed):
board testing PASS for latest changes

peterwangsz committed:
boards: frdm_mcxa166, frdm_mcxa276: add dma support

1. enable dma support

Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>
